src/Rules/Operators/OperandsInTernaryOperatorRule.php

```diff
@@ -0,0 +1,50 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Operators;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Ternary;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\ConstantType;
+use PHPStan\Type\VerbosityLevel;
+
+class OperandsInTernaryOperatorRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return Ternary::class;
+	}
+
+	/**
+	 * @param Node $node
+	 * @param Scope $scope
+	 * @return string[]
+	 */
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$node instanceof Ternary) {
```

Comment:
Unnecessary check as we are only processing `Ternary` nodes given by the `getNodeType` method return